packages/paypal-commerce-integration/src/paypal-commerce-credit-card/paypal-commerce-credit-cards-payment-strategy.ts

```diff
@@ -425,15 +425,19 @@ export default class PayPalCommerceCreditCardsPaymentStrategy implements Payment
         const billingAddress = state.getBillingAddressOrThrow();
 
         const submitConfig = {
-            billingAddress: {
-                company: billingAddress.company,
-                addressLine1: billingAddress.address1,
-                addressLine2: billingAddress.address2,
-                adminArea1: billingAddress.stateOrProvinceCode,
-                adminArea2: billingAddress.city,
-                postalCode: billingAddress.postalCode,
-                countryCode: billingAddress.countryCode,
-            },
+            ...(this.isCreditCardVaultedForm
+                ? {}
+                : {
+                      billingAddress: {
+                          company: billingAddress.company,
+                          addressLine1: billingAddress.address1,
+                          addressLine2: billingAddress.address2,
+                          adminArea1: billingAddress.stateOrProvinceCode,
+                          adminArea2: billingAddress.city,
+                          postalCode: billingAddress.postalCode,
+                          countryCode: billingAddress.countryCode,
+                      },
+                  }),
```

Comment:
Instead of changing submitConfig to an empty object, it might be better to does not send config on `await cardFields.submit(submitConfig);` level.

So in your case you will have:

```
if (this.isCreditCardVaultedForm) {
    await cardFields.submit();
} else {
    await cardFields.submit(submitConfig);
}
```

Reply:
Done